File: app/src/main/java/com/example/kirchhoff/example/ui/recycler/swipe/DataAdapter.java
package com.example.kirchhoff.example.ui.recycler.swipe;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.kirchhoff.example.R;

import java.util.ArrayList;

/**
 * @author Kirchhoff-
 */

public class DataAdapter extends RecyclerView.Adapter<DataAdapter.ViewHolder> {

    private ArrayList<String> countries;


    public DataAdapter(ArrayList<String> countries) {
        this.countries = countries;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.r_recycler_custom,
                parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.bind(countries.get(position));
    }

    @Override
    public int getItemCount() {
        return countries.size();
    }

    public void addItem(String country) {
        countries.add(country);
        notifyItemInserted(countries.size());
    }

    public void removeItem(int position) {
        countries.remove(position);
        notifyItemRemoved(position);
        notifyItemRangeChanged(position, countries.size());
    }

    class ViewHolder extends RecyclerView.ViewHolder {

        TextView country;

        public ViewHolder(View itemView) {
            super(itemView);

            country = (TextView) itemView.findViewById(R.id.country);
        }

        public void bind(String text) {
            country.setText(text);
        }
    }
}

File: recyclerswipe/src/main/java/com/example/recyclerswipe/DataAdapter.kt
package com.example.recyclerswipe

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.kirchhoff.recyclerswipe.R
import java.util.*

class DataAdapter constructor(private val countries: ArrayList<String>) : RecyclerView.Adapter<DataAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(parent.context)
                .inflate(R.layout.r_recycler_custom, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(countries[position])
    }

    override fun getItemCount() = countries.size

    fun addItem(country: String) {
        countries.add(country)
        notifyItemInserted(countries.size)
    }

    fun removeItem(position: Int) {
        countries.removeAt(position)
        notifyItemRemoved(position)
        notifyItemRangeChanged(position, countries.size)
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private var country: TextView = itemView.findViewById(R.id.country)

        fun bind(text: String) {
            country.text = text
        }
    }
}
